src/lib/reasoners/intervalCalculus.ml

```diff
@@ -2113,32 +2141,33 @@ let optimizing_split env uf opt_split =
               Printer.print_dbg "%a is a %s bound of %a" Q.print optim
                 (if to_max then "upper" else "lower") Expr.print e
           end;
-        let r2 = alien_of (P.create [] optim  ty) in
+        let r2 = alien_of (P.create [] optim ty) in
         Debug.case_split r1 r2;
         let t2 = mk_const_term optim ty in
         let value =
           if is_le then
-            Th_util.Value t2
+            Objective.Value.Value t2
           else
             begin
-              if to_max then Th_util.Limit (Below, t2)
-              else Th_util.Limit (Above, t2)
+              if to_max then Objective.Value.Limit (Below, t2)
+              else Objective.Value.Limit (Above, t2)
             end
         in
-        let case_split =
+        let r2 =
           if is_le then
-            Some (LR.mkv_eq r1 r2, true, Th_util.CS (Th_util.Th_arith, Q.one))
+            r2
           else
             (* As some bounds are strict, the value [r2] doesn't satisfy the
-               constraints of the problem. We don't produce a case-split
-               in this case because we don't want to propagate the equality
-               [r1 = r2] in the CC(X) environment [gamma_finite]. But
-               we propagate the new value [value] for this objective. Indeed,
-               we want to be able to print it while using the statement
-               [get-objective]. *)
-            None
+               constraints of the problem. In this case, we pick a value
+               in the domain of the polynomial [p]. But we propagate the new
+               value [value] for this objective. Indeed, we want to be able
+               to print it while using the statement [get-objective]. *)
+            middle_value env ~to_max ty p None
```

Comment:
Same remark as above; are standard model generation mechanisms failing somehow?

Reply:
Before this PR, we use a wrapped case-split as input for this function and we have access to the last computed value for the first-order model through this wrapped case-split. 

An alternative implementation would be to give this value as an argument of `optimizing_objective`. 

Notice that my `middle_value` gives a value in the last interval (for the `to_max = true` case) of the domain of the polynomial `p`.  You told me other solvers try to give a bigger value when we maximize strict bound.

Reply:
Ah, right this is for that! Makes sense. What I was thinking for that case (and what other solvers do I believe) is add an inequality constraint rather than an equality in case that specific value end up not being feasible. For instance, with a nonlinear objective `x * x` with `x in `1 .. 2` range, the range for `x * x` is `1 .. 4` (AFAIK Alt-Ergo know this) but if we pick `2` or `3` as a value there is no model whereas if we just constrain `< 3` or `> 2` it is.

(I think in this specific case we might go into an infinite loop in the current PR if we actually used this split in `look_for_sat`, not sure)

Reply:
So I suggest to keep the code as it's now. We could make better change later.

Reply:
Sure I am fine with keeping that code as is for now, but the code in `look_for_sat` needs to be fixed to actually use the split during model generation!